influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -372,10 +386,63 @@ impl ClosedBufferSegment {
                             sort_key: vec![],
                         };
 
+                        // All of the record batches for this table that we will
+                        // want to dedupe
+                        let batches = table_buffer.record_batches(table.schema());
+                        let row_count = batches.iter().map(|b| b.num_rows()).sum();
+
+                        // Dedupe and sort using the COMPACT query built into
+                        // iox_query
+                        let mut chunks: Vec<Arc<dyn QueryChunk>> = vec![];
                         let time_min_max = table_buffer.timestamp_min_max();
+                        let schema = table.schema();
+
+                        let chunk_stats = create_chunk_statistics(
+                            Some(row_count),
+                            schema,
+                            Some(time_min_max),
+                            None,
+                        );
 
-                        // persist every table buffer
-                        let data = table_buffer.record_batches(table.schema());
+                        chunks.push(Arc::new(BufferChunk {
+                            batches,
+                            schema: schema.clone(),
+                            stats: Arc::new(chunk_stats),
+                            partition_id: TransitionPartitionId::new(
+                                TableId::new(0),
+                                &self.segment_key,
+                            ),
+                            sort_key: None,
```

Comment:
We should default the sort key to the lexicographically ordered tag names. I believe passing None into compact will make it only use time, which would be bad.

Reply:
so grab the tags from the table, sort them lexicographically and then pass them in to the sort key? I can do that.

Reply:
FWIW the distributed product chooses a different sort order per-partition (though they are all "compatible"). If you allow new tags to arrive in other partitions/chunks you'll need to ensure that you can merge them correctly together (aka that the sorts are compatible)

Reply:
yeah, this is a temporary thing. We're going to sort by _series_id ;)

Reply:
😬